cmd/server.go

```diff
@@ -127,13 +123,15 @@ var serverCmd = &cobra.Command{
 			}
 
 			if err != nil {
-				httpLog.WithError(err).Fatal("failed to serve")
+				return fmt.Errorf("%v: failed to serve", err)
 			}
-
 			httpLog.Info("halted")
-		}()
+			return
+		})
 
-		running.Wait()
+		if err = group.Wait(); err != nil {
+			log.Fatal(err)
```

Comment:
Could you change this to `log.WithError(err).Fatal("appropriate message")`? Maybe something like "serving failed"?